Doc/tools/check-warnings.py

```diff
@@ -24,28 +26,150 @@
     "venv",
 }
 
-PATTERN = re.compile(r"(?P<file>[^:]+):(?P<line>\d+): WARNING: (?P<msg>.+)")
+# Regex pattern to match the parts of a Sphinx warning
+WARNING_PATTERN = re.compile(
+    r"(?P<file>([A-Za-z]:[\\/])?[^:]+):(?P<line>\d+): WARNING: (?P<msg>.+)"
+)
 
+# Regex pattern to match the line numbers in a Git unified diff
+DIFF_PATTERN = re.compile(
+    r"^@@ -(?P<linea>\d+),(?P<removed>\d+) \+(?P<lineb>\d+),(?P<added>\d+) @@",
+    flags=re.MULTILINE,
+)
 
-def check_and_annotate(warnings: list[str], files_to_check: str) -> None:
+
+def get_diff_files(ref_a: str, ref_b: str, filter_mode: str = "") -> set[Path]:
+    """List the files changed between two Gif refs, filtered by change type."""
+    added_files_result = subprocess.run(
+        [
+            "git",
+            "diff",
+            f"--diff-filter={filter_mode}",
+            "--name-only",
+            f"{ref_a}...{ref_b}",
+            "--",
+        ],
+        stdout=subprocess.PIPE,
+        check=True,
+        text=True,
+    )
+
+    added_files = added_files_result.stdout.strip().split("\n")
+    return {Path(file.strip()) for file in added_files if file.strip()}
+
+
+def get_diff_lines(ref_a: str, ref_b: str, file: os.PathLike) -> list[int]:
+    """List the lines changed between two Gif refs for a specific file."""
+    diff_output = subprocess.run(
+        [
+            "git",
+            "diff",
+            "--unified=0",
+            "--diff-filter=M",
```

Comment:
Perhaps add a comment explaining the diff filter used?
```suggestion
            # https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-filterACDMRTUXB82308203
            "--diff-filter=M",
```

Reply:
Actually, its redundant for this call, as we're explicitly passing in a specific modified file to get the diffs on (it was a holdover from the other function, where I explicitly describe it), so I simply removed it in my last cleanup commit.